### PR TITLE
Docs: Add User Operator docs

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Documentation

### Description

This PR adds documentation related to the User Operator. After #742 is merged, it might need to change into whcih sections are the assemblies included. But it should be ready for review as the test is final.

### Checklist

- [x] Update documentation

### CODE REVIEW

File: documentation/book/proc-creating-kafka-user.adoc [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+= Creating a user
+
+.Prerequisites
+
+* A running Kafka cluster.
+* A running User Operator.
+
+.Procedure
+
+. Prepare a file containing the `KafkaUserq to be created.
```
Comment (ppatierno) [inline documentation/book/proc-creating-kafka-user.adoc]: Typo "KafkaUserq ...

File: documentation/book/proc-deploying-the-topic-operator-standalone.adoc [hunk 2]
```diff
@@ -27,13 +27,13 @@ ifdef::Kubernetes[]
 On {KubernetesName} this can be done using `kubectl apply`:
 +
 [source,shell]
-kubectl apply -f examples/install/cluster-operator
+kubectl apply -f examples/install/user-operator
```
Comment (ppatierno) [inline documentation/book/proc-deploying-the-topic-operator-standalone.adoc]: the file is about topic-operator standalone why user-operator here?

File: documentation/book/proc-deploying-the-topic-operator-standalone.adoc [hunk 3]
```diff
@@ -27,13 +27,13 @@ ifdef::Kubernetes[]
...
 +
 endif::Kubernetes[]
 On {OpenShiftName} this can be done using `oc apply`:
 +
 [source,shell]
-oc apply -f examples/install/cluster-operator
+oc apply -f examples/install/user-operator
```
Comment (ppatierno) [inline documentation/book/proc-deploying-the-topic-operator-standalone.adoc]: Ditto as above

File: documentation/book/proc-deploying-the-user-operator-standalone.adoc [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+.Prerequisites
+
+* An existing Kafka cluster for the User Operator to connect to.
+
+.Procedure
+
+. Edit the `examples/install/user-operator/05-Deployment-strimzi-user-operator.yaml` resource. You will need to change the following
++
+.. The `STRIMZI_CA_NAME` environment variable in `Deployment.spec.template.spec.containers[0].env` should be set to point to an {ProductPlatformName} Secret which should contain the Cetificate Authority for signing new user certificates for TLS Client Authentication.
+The secret should contain the public key of the certificate authority under the key `clients-ca.crt and the pribate key under `clients-ca.key`.
```
Comment (ppatierno) [inline documentation/book/proc-deploying-the-user-operator-standalone.adoc]: missing an apex at the end of clients-ca.crt and typo in "pribate"

File: documentation/book/proc-deploying-the-topic-operator-standalone.adoc (hunk 2 above)
Comment (scholzj) [reply documentation/book/proc-deploying-the-topic-operator-standalone.adoc]: See? I try to fix something and fix it incorrectly :-(. I might need a vacation as well :-)

Comment (ppatierno) [reply documentation/book/proc-deploying-the-topic-operator-standalone.adoc]: You don't need vacation ... you deserve it! ;)

File: documentation/book/con-what-the-user-operator-does.adoc [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+The role of the User Operator is to watch for `KafkaUser` {ProductPlatformName} resources describing Kafka users and make sure they are configured inside the Kafka cluster as declared in the `KafkaUser` resource.
+
+Specifically:
+
+* if a `KafkaUser` is created, the operator will create the user it describes
+* if a `KafkaUser` is deleted, the operator will delete the user it describes
+* if a `KafkaUser` is changed, the operator will update the user it describes
+
+Unlike the xref:what-the-topic-operator-does-str[Topic Operator], the User Operator does not sync any changes back from the Kafka cluster into the {ProductPlatformName} resources.
```
[the remark this answers is not in the document]
Comment (scholzj) [reply documentation/book/con-what-the-user-operator-does.adoc]: I do not think these say the same ... one says that we are not doing something. And the other one explains why.

File: documentation/book/con-what-the-user-operator-does.adoc [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+Unlike the Kafka topics which might be created by applications directly in Kafka, it is not expected that the users will be managed directly in the Kafka cluster in parallel with the User Operator, so this should not be needed.
+
+User Operator allows you to declare a `KafkaUser` as part of your application's deployment and the User Operator will take care of creating the user for you.
+When the user is created, the credentials will be created in a secret.
```
[the remark this answers is not in the document]
Comment (scholzj) [reply documentation/book/con-what-the-user-operator-does.adoc]: I changed it to `Secret` to make it clear it refers to an OpenShift / Kubernetes resource type.

File: documentation/book/getting-started.adoc [hunk 7]
```diff
@@ -40,6 +40,7 @@ include::assembly-kafka-cluster.adoc[leveloffset=+1]
 include::assembly-kafka-connect.adoc[leveloffset=+1]
 
 include::assembly-hello-world.adoc[leveloffset=+1]
+
 //Modularized above here
```
[the remark this answers is not in the document]
Comment (scholzj) [reply documentation/book/getting-started.adoc]: I have done this in the other PR, so I will leave it here.

File: documentation/book/ref-kafka-user.adoc [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+Topic and Group resources additionally allow to specify the name of the resource for which the rule applies.
+The name can be specified in the `name` property.
+The name can be either specified as literal or as a prefix.
+To specify the name as literal, set the `patternType` property to the value `literal`.
+Literal names will be taken exactly as they are specified in the `name` field.
+To specify the name as a prefix, set the `patternType` property to the value `prefix`.
+Prefix type names will use the value from the `name` only a prefix and will apply the rule to all resources with names starting with the value.
+The cluster type resources have no name.
+
+For more details about `SimpleAclAuthorizer`, its ACL rules and the allowed combinations of resources and operations, visit link:http://kafka.apache.org/documentation/#security_authz[Authorization and ACLs^].
```
[the remark this answers is not in the document]
Comment (scholzj) [reply documentation/book/ref-kafka-user.adoc]: TBH, I'm not sure I understand this point: _Authorization and ACLs_ is the title of the target page.

File: documentation/book/ref-kafka-user.adoc [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+
+Authorization is configured using the `authorization` property in `KafkaUser.spec`.
+The authorization type enabled for this user will be specified using the `type` field.
+Currently, the only supported authorization type is the Simple authorization.
+
+When no authorization is specified, User Operator will not provision any access rights for the user.
+
+=== Simple Authorization
+
+To use Simple Authorization, set the `type` property to `simple`.
```
[the remark this answers is not in the document]
Comment (scholzj) [reply documentation/book/ref-kafka-user.adoc]: The authorization is part of the `KafkaUser` resource. The `proc-creating.kafka.user.adoc`, and `proc-changing-kafka-user.adoc` even contain an example with ACLs. There is no other procedure IMHO.

File: documentation/book/ref-kafka-user.adoc [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+[id='ref-kafka-user-{context}']
+= Kafka User resource
+
+The Kafka User resource is used to declare a user with its authentication mechanism, authorization mechanism and access rights.
+
+== Authentication
+
+Authentication is configured using the `authentication` property in `KafkaUser.spec`.
+The authentication mechanism enabled for this user will be specified using the `type` field.
+Currently, the only support authentication mechanism is the TLS Client Authentication mechanism.
```
[the remark this answers is not in the document]
Comment (scholzj) [reply documentation/book/ref-kafka-user.adoc]: Why do you think this is _noteworthy_? It just lists the supported mechanisms. If the don't notice it here, they will anyway find no other mechanism than the one described below.

**Pull request merged**